src/main/java/com/ait/lienzo/client/core/shape/TextTruncateWrapper.java

```diff
@@ -0,0 +1,239 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape;
+
+import java.util.ArrayList;
+
+import com.ait.lienzo.client.core.Context2D;
+import com.ait.lienzo.client.core.types.BoundingBox;
+
+/**
+ * ITextWrapper implementation that truncates text and appends "..." if there is no space left.
+ */
+@SuppressWarnings("Duplicates")
+public class TextTruncateWrapper extends TextNoWrap {
+
+    private BoundingBox wrapBoundaries;
+    private final double Y_OFFSET = 0.8;
+
+    public TextTruncateWrapper(final Text text,
+                               final BoundingBox wrapBoundaries) {
+        super(text);
+        setWrapBoundaries(wrapBoundaries);
+    }
+
+    public BoundingBox getWrapBoundaries() {
+        return wrapBoundaries;
+    }
+
+    public TextTruncateWrapper setWrapBoundaries(final BoundingBox boundaries) {
+        wrapBoundaries = boundaries;
+        return this;
+    }
+
+    @Override
+    public BoundingBox getBoundingBox() {
+        final double[] boundaries = calculateWrapBoundaries();
+        return new BoundingBox().addX(0).addX(wrapBoundaries.getWidth()).addY(0).addY(boundaries[1]);
+    }
+
+    private double getWrapBoundariesWidth() {
+        final double margin = 4 * getBoundingBoxForString(" ").getWidth();
+        return this.wrapBoundaries.getWidth() - margin;
+    }
+
+    private double[] calculateWrapBoundaries() {
+        final String[] words = textSupplier.get().split("\\s");
+        if (words.length < 1) {
+            return new double[]{wrapBoundaries.getX(), wrapBoundaries.getY()};
+        }
+
+        final double wrapWidth = getWrapBoundariesWidth();
+        final String firstWord = words[0];
+        double width = getBoundingBoxForString(firstWord).getWidth();
+        final StringBuilder nextLine = new StringBuilder(firstWord);
+        int numOfLines = 1;
+        for (int i = 1; i < words.length; i++) {
+            width = getBoundingBoxForString(nextLine + " " + words[i]).getWidth();
+            if (width <= wrapWidth) {
+                nextLine.append(" ").append(words[i]);
+            } else {
+                nextLine.setLength(words[i].length());
+                nextLine.replace(0,
+                                 words[i].length(),
+                                 words[i]);
+                numOfLines++;
+            }
+        }
+
+        final double lineHeight = (getBoundingBoxForString(" ").getHeight());
+
+        while (!hasVerticalSpace(numOfLines, lineHeight, wrapBoundaries.getHeight() - (Y_OFFSET * numOfLines))
```

Comment:
Why can we not calculate the maximum row number rather than iterate here?

Row count would be `(wrapBoundaries.getHeight() - margin * 2) / lineHeight`?

Reply:
I did this way at the first time, but it not worked, because of the Y_OFFSET.
So in some scenarios the text is not vertical aligned and lays closer to the top.
What happens is that the first estimative is not that precise (because the algorithm to do a precise estimative is costly) and this while ensures that we are not overflowing the height considering the Y_OFFSET. 
About the `Y_OFFSET`, this is used to vertical centre align the text, internally (I had to dig in Lienzo to found out why the text was not being correctly centre aligned).
And the offset depends on the current number of lines.